hl.vep gives AssertionError

Posting with recommendation from @konradjk 
### Hail version:
devel-2c596b7

### What you did:
Creating a ClinVar matrixtable from a tsv and vep'ing

```
import hail as hl
import hail.expr.aggregators as agg
from gnomad_hail import *
hl.init()

clinvar_ht= hl.import_table("gs://gnomad-resources/clinvar/source/clinvar_alleles.single.b37.tsv.gz", impute=True, missing='NA')
clinvar_ht = clinvar_ht.annotate(locus = hl.locus(clinvar_ht.chrom, clinvar_ht.pos),
                                 alleles = hl.array({clinvar_ht.ref, clinvar_ht.alt}))

clinvar_ht = clinvar_ht.key_by(clinvar_ht.locus, clinvar_ht.alleles)
clinvar_mt = hl.MatrixTable.from_rows_table(clinvar_ht)
clinvar_mt = split_multi_dynamic(clinvar_mt, left_aligned = False)
clinvar_mt = clinvar_mt.repartition(100)
clinvar_mt = clinvar_mt.key_rows_by(clinvar_mt.locus, clinvar_mt.alleles)
clinvar_vep = hl.vep(clinvar_mt, vep_config)
```

### What went wrong (all error messages here, including the full java stack trace):
```
In [6]: clinvar_vep = hl.vep(clinvar_mt, vep_config)
2018-03-08 02:46:03 Hail: WARN: property `hail.vep.assembly' not specified. Setting to GRCh37
[Stage 22:======================================================>(99 + 1) / 100]2018-03-08 02:54:37 Hail: INFO: vep: annotated 243477 variants
---------------------------------------------------------------------------
FatalError                                Traceback (most recent call last)
<ipython-input-6-a229f1f9de81> in <module>()
----> 1 clinvar_vep = hl.vep(clinvar_mt, vep_config)

<decorator-gen-843> in vep(dataset, config, block_size, name, csq)

/hadoop_gcs_connector_metadata_cache/hail/hail-devel-0c961806173f.zip/hail/typecheck/check.py in _typecheck(__orig_func__, *args, **kwargs)
    491     def _typecheck(__orig_func__, *args, **kwargs):
    492         args_, kwargs_ = check_all(__orig_func__, args, kwargs, checkers, is_method=False)
--> 493         return __orig_func__(*args_, **kwargs_)
    494
    495     return decorator(_typecheck)

/hadoop_gcs_connector_metadata_cache/hail/hail-devel-0c961806173f.zip/hail/methods/qc.py in vep(dataset, config, block_size, name, csq)
    545
    546     require_row_key_variant(dataset, 'vep')
--> 547     return MatrixTable(Env.hail().methods.VEP.apply(dataset._jvds, config, 'va.`{}`'.format(name), csq, block_size))
    548
    549

/usr/lib/spark/python/lib/py4j-0.10.4-src.zip/py4j/java_gateway.py in __call__(self, *args)
   1131         answer = self.gateway_client.send_command(command)
   1132         return_value = get_return_value(
-> 1133             answer, self.gateway_client, self.target_id, self.name)
   1134
   1135         for temp_arg in temp_args:

/hadoop_gcs_connector_metadata_cache/hail/hail-devel-0c961806173f.zip/hail/utils/java.py in deco(*args, **kwargs)
    236             # this is a hack to suppress the original error's stack trace
    237             if _exception:
--> 238                 raise _exception
    239
    240     return deco

FatalError: AssertionError: assertion failed

Java stack trace:
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:156)
	at is.hail.variant.MatrixTable.orderedRVDLeftJoinDistinctAndInsert(MatrixTable.scala:982)
	at is.hail.methods.VEP$.annotate(VEP.scala:429)
	at is.hail.methods.VEP$.apply(VEP.scala:434)
	at is.hail.methods.VEP.apply(VEP.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)

Hail version: devel-2c596b7
Error summary: AssertionError: assertion failed
```

--- a/src/main/scala/is/hail/methods/Nirvana.scala
+++ b/src/main/scala/is/hail/methods/Nirvana.scala
@@ -223,6 +223,7 @@ object Nirvana {
   }
 
   def annotate(vds: MatrixTable, config: String, blockSize: Int, root: String = "va.nirvana"): MatrixTable = {
+    assert(vds.rowKey == IndexedSeq("locus", "alleles"))
     val parsedRoot = Parser.parseAnnotationRoot(root, Annotation.VARIANT_HEAD)
 
     val properties = try {
@@ -320,7 +321,7 @@ object Nirvana {
     info(s"nirvana: annotated ${ annotations.count() } variants")
 
     val nirvanaORVDType = new OrderedRVDType(
-      Array("locus"), Array("locus", "alleles"),
+      vds.rowPartitionKey.toArray, vds.rowKey.toArray,
       TStruct(
         "locus" -> vds.rowKeyTypes(0),
         "alleles" -> vds.rowKeyTypes(1),

--- a/src/main/scala/is/hail/methods/VEP.scala
+++ b/src/main/scala/is/hail/methods/VEP.scala
@@ -217,6 +217,7 @@ object VEP {
 
   def annotate(vsm: MatrixTable, config: String, root: String = "va.vep", csq: Boolean,
     blockSize: Int): MatrixTable = {
+    assert(vsm.rowKey == IndexedSeq("locus", "alleles"))
 
     val parsedRoot = Parser.parseAnnotationRoot(root, Annotation.VARIANT_HEAD)
 
@@ -396,7 +397,7 @@ object VEP {
     val vepType: Type = if (csq) TArray(TString()) else vepSignature
 
     val vepORVDType = new OrderedRVDType(
-      Array("locus"), Array("locus", "alleles"),
+      vsm.rowPartitionKey.toArray, vsm.rowKey.toArray,
       TStruct(
         "locus" -> vsm.rowKeyTypes(0),
         "alleles" -> vsm.rowKeyTypes(1),

